cmake/FindMKL.cmake

```diff
@@ -0,0 +1,78 @@
+###########################################################
+# Adapted from PRACE course materials at:
+# https://repository.prace-ri.eu/git/CodeVault/hpc-kernels/spectral_methods/-/blob/master/cmake/Modules/FindMKL.cmake
+###########################################################
+
+# This script looks for MKL in two locations:
+# - The environment variable MKLROOT, which is defined by
+#    sourcing an MKL environment
+# - The directory `/opt/intel/mkl`, which is a common
+#    install location for MKL.
+# It may be possible to install MKL using python -m pip,
+# though this is not guaranteed to be found, and may
+# require explicitly setting the MKLROOT variable.
+
+###########################################################
+# Stage 1: find the root directory
+###########################################################
+
+set(MKLROOT_PATH $ENV{MKLROOT})
+
+if (NOT MKLROOT_PATH)
+  if (EXISTS "/opt/intel/mkl")
+    set(MKLROOT_PATH "/opt/intel/mkl")
+  endif ()
+endif ()
```

Comment:
I wonder if this now needs updating with the change in oneAPI's MKL release. Maybe we should add `/opt/intel/oneapi/mkl/latest/` or similar to the paths here, or just replace the old path with this

Reply:
Ok, made this more robust. I installed OneAPI, tested the options and everything seems in order now.